crates/runtime/src/actor/component/config.rs

```diff
@@ -0,0 +1,35 @@
+use super::{Ctx, Instance};
+
+use crate::capability::config::{self, runtime};
+use crate::capability::Config;
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use tracing::instrument;
+
+impl Instance {
+    /// Set [`Config`] handler for this [Instance].
+    pub fn config(&mut self, config: Arc<dyn Config + Send + Sync>) -> &mut Self {
+        self.handler_mut().replace_config(config);
+        self
+    }
+}
+
+#[async_trait]
+impl runtime::Host for Ctx {
+    #[instrument(skip(self))]
+    async fn get(
+        &mut self,
+        key: String,
+    ) -> anyhow::Result<Result<Option<String>, config::runtime::ConfigError>> {
```

Comment:
I know the runtime interface is implemented as a `Result<Option<String>>`, is there a reason why we double wrap this here with another anyhow Result?

Reply:
The result is something defined by the host bindings insofar as I understand. I think everything is wrapped by a result type